src/test/java/asserts/TestCaseAssert.java

```diff
@@ -10,49 +16,125 @@
         softAssert.assertAll();
     }
 
     public static void updateTestCaseAssert(TestCaseResponse actualResponse, TestCaseResponse expectedResponse) {
+        if (actualResponse == null) {
+            Assert.fail("Actual test case is null");
+        }
+        if (actualResponse.getTestSteps() == null) {
+            Assert.fail("Actual test step is null");
+        }
+        if (actualResponse.getTestSteps().size() != expectedResponse.getTestSteps().size()) {
+            Assert.fail("The number of test steps is not the same in actual and expected responses");
+        }
         SoftAssert softAssert = new SoftAssert();
         softAssert.assertEquals(actualResponse.getId(), expectedResponse.getId(), "id didn't match");
+        softAssert.assertNotNull(actualResponse.getCandidateId(), "candidate id is null");
         softAssert.assertEquals(actualResponse.getTitle(), expectedResponse.getTitle(), "title didn't match");
         softAssert.assertEquals(actualResponse.getExpectedResult(), expectedResponse.getExpectedResult(), "expected result didn't match");
         softAssert.assertEquals(actualResponse.getDescription(), expectedResponse.getDescription(), "description didn't match");
+        softAssert.assertEquals(actualResponse.getAutomated(), expectedResponse.getAutomated(), "automated didn't match");
+        softAssert.assertNotNull(actualResponse.getCandidateScenarioId(), "scenario id is null");
 
-        List<TestStepRequest> testStepRequestList = expectedResponse.getTestSteps();
+        List<TestStepResponse> testStepRequestList = expectedResponse.getTestSteps();
         for(int i = 0; i < testStepRequestList.size(); i++) {
+            softAssert.assertNotNull(actualResponse.getTestSteps().get(i).getId(), "test step id is null");
+            softAssert.assertNotNull(actualResponse.getTestSteps().get(i).getTestcaseId(), "test case id in test step is null");
             softAssert.assertEquals(actualResponse.getTestSteps().get(i).getValue(), expectedResponse.getTestSteps().get(i).getValue(), "test steps didn't match");
         }
+        softAssert.assertAll();
+    }
+
+    public static void deleteTestCase(ApiError actualError, ApiError expectedError) {
+        SoftAssert softAssert = new SoftAssert();
+        softAssert.assertEquals(actualError.getError(), expectedError.getError(), "error message is not the same");
+        softAssert.assertAll();
+    }
+
+    public static void DeleteSuccessMessage(DeleteTestCaseResponse actualResponse, DeleteTestCaseResponse expectedResponse) {
+        SoftAssert softAssert = new SoftAssert();
+        softAssert.assertEquals(actualResponse.getSuccess(), expectedResponse.getSuccess(), "success message didn't match");
+        softAssert.assertAll();
+    }
+
+    public static void createTestCaseWithoutRequiredField(ApiRequiredFieldError actualError, ApiRequiredFieldError expectedError) {
+        SoftAssert softAssert = new SoftAssert();
```

Comment:
nemaš proveru da actual nije null

Reply:
Da li moze ovako:
  if (actualError == null) {
            Assert.fail("Actual error is null");
        }